bundles/azure/src/main/java/org/apache/gravitino/abs/fs/AzureFileSystemProvider.java

```diff
@@ -58,12 +64,42 @@ public FileSystem getFileSystem(@Nonnull Path path, @Nonnull Map<String, String>
           config.get(AzureProperties.GRAVITINO_AZURE_STORAGE_ACCOUNT_KEY));
     }
 
-    if (!config.containsKey(ABFS_IMPL_KEY)) {
+    if (!hadoopConfMap.containsKey(ABFS_IMPL_KEY)) {
       configuration.set(ABFS_IMPL_KEY, ABFS_IMPL);
     }
 
     hadoopConfMap.forEach(configuration::set);
 
+    if (enableGravitinoCredentialVending(hadoopConfMap)) {
+      try {
+        AzureSasCredentialsProvider azureSasCredentialsProvider = new AzureSasCredentialsProvider();
```

Comment:
This seems hacky and hard to understand
```
        AzureSasCredentialsProvider azureSasCredentialsProvider = new AzureSasCredentialsProvider();
        azureSasCredentialsProvider.initialize(configuration, null);
        String sas = azureSasCredentialsProvider.getSASToken(null, null, null, null);
        if (sas != null) { xx } else {xx}
```

Reply:
I think using `GravitinoFileSystemCredentialsProvider` to get Gravitino credentail directly is also a good solutions to this problem.